src/Data/Vec/Properties.agda

```diff
@@ -472,6 +541,20 @@ toList-map : ∀ (f : A → B) (xs : Vec A n) →
 toList-map f [] = refl
 toList-map f (x ∷ xs) = cong (f x List.∷_) (toList-map f xs)
 
+truncate-map : (f : A → B) (m : ℕ) (m≤n : m ≤ n) (xs : Vec A n) →
+  map f (truncate m≤n xs) ≡ truncate m≤n (map f xs)
+truncate-map {n = n} f m m≤n xs =
+  begin
+    map f (truncate m≤n xs)     ≡⟨ cong (map f) (truncate≡take m≤n xs eq) ⟩
+    map f (take m (cast eq xs)) ≡⟨ take-map f m _ ⟨
+    take m (map f (cast eq xs)) ≡⟨ cong (take m) (map-cast f eq xs) ⟩
+    take m (cast eq (map f xs)) ≡⟨ truncate≡take m≤n (map f xs) eq ⟨
+    truncate m≤n (map f xs) ∎
+ where
+   .eq : n ≡ m + (n ∸ m)
```

Comment:
No need to mark `eq` as irrelevant; it's the proof which is ignored by `cast` etc.
```suggestion
   eq : n ≡ m + (n ∸ m)
```